File: ch01_recursion/indirect-recursion.js
// function fun1(n){
//     if(n>0){
//         console.log(n)
//         fun2(n-1)
//     }
// }
// function fun2(n){
//     if(n>0){
//         console.log(n)
//         fun1(n/2)
//     }
// }
// fun1(20)
function fun(n){
    if(n>100){
        return n-10
    } else{
        return fun(fun(n+11))
    }
    return 0
}
fun(95)

File: abdul-beri/using_java/ch01_recursion_java/IndirectRecursion.java
package ch01_recursion_java;

public class IndirectRecursion {
    static void fun1(int n){
        if(n>0){
            System.out.print(" "+ n +" ");
            fun2(n-1);
        }
    }
    static void fun2(int n){
        if(n>1){
            System.out.print(" "+ n +" ");
            fun1(n/2);
        }
    }
    public static void main(String[] args) {
        int x = 20;
        fun1(x);
    }
}
